src/app/StorybookLayout.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { composeStory } from '@storybook/react';
+
+import * as stories from '../page-stories/build.stories';
+import { cookies, headers } from 'next/headers';
+import { Mock } from '../mock';
+
+const map = {
+  'pages-build--build-seven-desc': {
+    csf: stories,
+    key: 'BuildSevenDesc',
+  },
+  'pages-build--build-six': {
+    csf: stories,
+    key: 'BuildSix',
+  },
+};
+
+export const StorybookLayout = ({ children }) => {
+  const storyCookie = cookies().get('__storyId__');
```

Comment:
I had to use a cookie because you cannot read query params ("search params") in layouts AFAICT.

Perhaps there's some other way to run code on every request in Next. Maybe the redirect wasn't necessary and I could just do it in the middleware?